### ISSUE TITLE
False positive too-many-positional-arguments on Protocols with ParamSpec for Python >= 3.12

### ISSUE THREAD

**Issue by sibarras** [astral-sh/ty#474]
### Summary

when using a Protocol with ParamSpec in a class, ty doesn't allow to specify the ParamSpec for the protocol, and fails saying that the protocol is not expecting arguments.
see:
```python3
from typing import Protocol


class MyClass[**P](Protocol):
    def return_arg(self, *args: P.args, **kwargs: P.kwargs) -> None: ...


def my_func[**P2](x: MyClass[P2], *args: P2.args, **kwargs: P2.kwargs) -> None:  #  Too many positional arguments to class `MyClass`: expected 0, got 1 ty[too-many-positional-arguments](https://ty.dev/rules#too-many-positional-arguments)
    return x.return_arg(*args, **kwargs)

```

### Version

ty 0.0.1-alpha.6

**Comment by IbarraSamuel** [comment]
Simplified version.
```python3
from typing import Protocol


class MyClass[**P](Protocol):
    pass


def my_func[**P2](_x: MyClass[P2]):
    pass
```

**Comment by carljm** [comment]
Thanks for the report. We don't support `ParamSpec` yet, but we should soon, which will fix this.

**Comment by carljm** [comment]
Closing as duplicate of #157 -- thanks for the report!
